astra/src/main/java/com/slack/astra/logstore/search/LogIndexSearcher.java

```diff
@@ -12,5 +13,7 @@ SearchResult<T> search(
       Long maxTime,
       int howMany,
       AggBuilder aggBuilder,
-      QueryBuilder queryBuilder);
+      QueryBuilder queryBuilder,
+      AstraSearch.SearchRequest.FieldInclusion includeFields,
```

Comment:
You'll notice that we don't actually pass the raw proto objects around in the code, and only use these for the wire communication. These get turned into internal objects with a combination of from/to methods (ie, see `SearchResultUtils.fromSearchRequest` for an example).

This is generally preferred, because it allows you to iterate and change the communication mechanism without having to change the internal data structures. It also opens up the opportunity to better test those internal method calls, and to re-use them later in other ways that aren't tied to the original request object.